apps/bfd-pipeline/bfd-pipeline-shared-utils/src/main/java/gov/cms/bfd/pipeline/sharedutils/s3/AwsS3ClientFactory.java

```diff
@@ -0,0 +1,42 @@
+package gov.cms.bfd.pipeline.sharedutils.s3;
+
+import gov.cms.bfd.pipeline.sharedutils.S3ClientConfig;
+import gov.cms.bfd.sharedutils.config.AwsClientConfig;
+import lombok.AllArgsConstructor;
+import software.amazon.awssdk.awscore.defaultsmode.DefaultsMode;
+import software.amazon.awssdk.services.s3.S3AsyncClient;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.S3ClientBuilder;
+import software.amazon.awssdk.services.s3.S3CrtAsyncClientBuilder;
+
+/**
+ * Implementation of {@link S3ClientFactory} that creates real S3 clients based on a {@link
+ * AwsClientConfig} provided in it constructor.
+ */
+@AllArgsConstructor
+public class AwsS3ClientFactory implements S3ClientFactory {
+  /**
+   * Passed to {@link S3CrtAsyncClientBuilder#minimumPartSizeInBytes} to cause large files to be
+   * downloaded in parts of 200 mb.
+   */
+  public static final long MINIMUM_PART_SIZE_FOR_DOWNLOAD = 200L * 1024L * 1024L;
+
+  /** Used to configure the S3 client builders with basic connection settings. */
+  private final S3ClientConfig s3ClientConfig;
+
+  @Override
+  public S3Client createS3Client() {
```

Comment:
External to this work, but we can check the usages of this and see if it'd be beneficial to just use the async client in places at some point in the future; we might get some gains out of the multi part download in other places we download files.